### PR TITLE
Spc 788 reuse mocks on interface

### PR DESCRIPTION
**Pull request by BrianVanB**
There now is a single mock for keycloak in the react/\_\_mocks\_\_ folder.
In the tests you no longer need to call `jest.mock()` as jest will default to the mock.
If you want to change mock functionality you can call  `require('@react-keycloak/web').__someFunction`
See \_\_mocks\_\_/@react-keycloak/web.js for the available functions.

### CODE REVIEW

File: Interface/Dockerfile [hunk 1]
```diff
@@ -42,6 +42,7 @@ FROM testing AS unit-testing
 WORKDIR /react
 
 COPY ["react/testing/unit testing","./testing/unit testing"]
+COPY ["react/__mocks__", "./__mocks__"]
```
Comment (RienkF) [inline Interface/Dockerfile:45]: Why is this folder marked with underscores? I thought that was a python thing

File: Interface/react/testing/unit testing/app.test.tsx [hunk 2]
```diff
@@ -7,99 +7,90 @@ Utrecht University within the Software Project course.
...
+  await act(async () => {
+    render(<App />)
+  })
 })
 
-/** Test to simply check if the page renders */
-test('App renders without errors', async () => {
+//Start with testing what happens when keycloak has not finished loading yet
+test('waits for keycloak to load before showing any contents', async () => {
+  require('@react-keycloak/web').__SetMockInitialized(false)
```
Comment (RienkF) [inline Interface/react/testing/unit testing/app.test.tsx:26]: Also, can't you use an import for these as well?

File: Interface/Dockerfile (hunk 1 above)
Comment (BrianVanB) [reply Interface/Dockerfile:45]: Named according to documentation https://jestjs.io/docs/manual-mocks

File: Interface/react/testing/unit testing/app.test.tsx (hunk 2 above)
Comment (BrianVanB) [reply Interface/react/testing/unit testing/app.test.tsx:26]: Also followed examples from the documentation at https://jestjs.io/docs/manual-mocks

File: Interface/Dockerfile (hunk 1 above)
Comment (RienkF) [reply Interface/Dockerfile:45]: Ok, seems like it is indeed neccecary. Don't like it but it is what it is

File: Interface/react/testing/unit testing/app.test.tsx (hunk 2 above)
Comment (RienkF) [reply Interface/react/testing/unit testing/app.test.tsx:26]: Why they have to do us like this

**Review by RienkF — APPROVED** [review]

**Review by JTeeuwissen — APPROVED** [review]

**Pull request merged**